src/ansiblelint/output.py

```diff
@@ -43,63 +42,367 @@
+
+
+class Markdown(UserString):
+    """Markdown string."""
+
+    def display(self) -> None:
+        """Display markdown text in the terminal using an external renderer if available."""
+        global md_cmd  # pylint: disable=global-statement
+        if md_cmd is None:
+            for v in md_renderers:
+                if shutil.which(v):
+                    md_cmd = v
+                    break
+            if not md_cmd:
+                msg = f"No know markdown renderer found ({', '.join(md_renderers)}), output as plain text."
+                _logger.warning(msg)
+                md_cmd = ""
+            else:
+                msg = f"Using markdown renderer: {md_cmd}"
+                _logger.info(msg)
+
+        if md_cmd:
+            subprocess.run(  # noqa: S603
+                md_renderers[md_cmd],
+                input=self.data,
+                text=True,
+                check=False,
+            )
+        else:
+            console.print(self.data)
 
-def reconfigure(new_options: dict[str, Any]) -> None:
+
+color: bool = True
+
+
+# https://peps.python.org/pep-3101/
+__all__ = ("Console", "color")
+
+
+class Console:
+    """Console."""
+
+    colored: bool = True
+    style: type[PlainStyle] = AnsiStyle
+
+    def __init__(self, file: TextIO | None = sys.stdout):
+        """Console constructor."""
+        self._file = file
+
+    def print(
+        self,
+        *values: Any,
+        sep: str | None = " ",
+        end: str | None = "\n",
+        file: TextIO | None = None,
+        flush: bool = False,
+    ) -> None:
+        """Internal print implementation."""
+        buffer = StringIO()
+        print(*values, sep=sep, end="", file=buffer, flush=True)
+        buffer.seek(0)
+        data = buffer.read()
+        print(self.render(data), end=end, file=file or self._file, flush=flush)
+
+    def render(self, text: str) -> str:
+        """Parses a string containing nested BBCode with a generic block terminator ([/])."""
+        style: type[PlainStyle] = AnsiStyle if self.colored else PlainStyle
+        # Define bbcode-to-ansi mappings
+        bbcode_to_ansi = {
+            "bold": (style.bold, style.normal),
+            "dim": (style.dim, style.normal),
+            # logging
+            "warning": (style.warning, style.normal),
+            "error": (style.error, style.normal),
+            "info": (style.info, style.normal),
+            "debug": (style.debug, style.normal),
+            "noteset": (style.notset, style.normal),
+            # data types
+            "repr.path": (style.path, style.normal),
+            "repr.number": (style.number, style.normal),
+            "repr.link": (style.link, style.normal),
+            "failed": (style.failed, style.normal),
+            "success": (style.success, style.normal),
+        }
+        # Regex to find opening tags and their content
+        tag_pattern = re.compile(r"\[([\w\.]+)(?:=(.*?))?\]|\[/\]")
```

Comment:
I've found putting the regex and the use of in it's own fn to make testing easier and demostrate what it does in the unit tests.